packages/core/orchestration/src/transaction/transaction-orchestrator.ts

```diff
@@ -696,12 +700,28 @@ export class TransactionOrchestrator extends EventEmitter {
 
       if (!step.isCompensating()) {
         if (
-          step.definition.continueOnPermanentFailure &&
+          (step.definition.continueOnPermanentFailure ||
+            step.definition.skipOnPermanentFailure) &&
           !TransactionTimeoutError.isTransactionTimeoutError(timeoutError!)
         ) {
-          for (const childStep of step.next) {
-            const child = flow.steps[childStep]
-            child.changeState(TransactionStepState.SKIPPED_FAILURE)
+          if (step.definition.skipOnPermanentFailure) {
+            const until = isString(step.definition.skipOnPermanentFailure)
+              ? step.definition.skipOnPermanentFailure
+              : undefined
+
+            let stepsToSkip: string[] = [...step.next]
+            while (stepsToSkip.length > 0) {
+              const currentStep = flow.steps[stepsToSkip.pop()!]
+
+              if (until && currentStep.definition.action === until) {
```

Comment:
q: how would this work with parallelize(step1, step2, step3)?

Reply:
At the Transaction Orchestrator level, it will skip all the steps that are children of the step that has failed.
```
[
  a - a.1 - a.2
  b - b.1 - b.2 - b.3
  c - c.1
]
```
if `b` fails, `b.1, b.2, b.3` will be skipped, but the rest will continue.

For our workflow SDK that has a limitation where you can't have multiple next steps for each "branch" parallelized, next steps are only append as child of the last step defined on the `parallelize` 

```
[
  a,
  b,
  c - c.1 - c.2 - c.3
]
```
If `a` or `b` fail, nothing will be skipped, only if `c` fail it will skip the children steps.